src/inverted_index.c

```diff
@@ -466,8 +480,10 @@ size_t InvertedIndex_WriteEntryGeneric(InvertedIndex *idx, IndexEncoder encoder,
 
   idx->lastId = docId;
   blk->lastId = docId;
-  ++blk->numDocs;
-  ++idx->numDocs;
+  if (!same_doc) {
+    ++blk->numDocs;
+    ++idx->numDocs;
```

Comment:
How will this effect other parts of the code? Do we assume anywhere that numDocs are the actual number of entries in the inverted index? How does GC handle and maintain this number now (for example)?

Reply:
GC is still operating on `NumericRangeTree` which, as before, has one or more `NumericRange`, only now their `InvertedIndex` entries may contain several values from the same doc.

Seems we might still have some assumption about numRecords and numDocs being equivalent:
Need to review function such as `removeRange` with the line
```
rv->numRecords -= temp->entries->numDocs;
```

Reply:
One change in `NumericRangeNode_Add`, which seems more related to performance than to to GC, is when splitting a `NumericRange`, which now might be delayed due to the right-hand side condition depending of `numDocs`, which might not be incremented per entry.
```
if (card * NR_CARD_CHECK >= n->range->splitCard || 
      (n->range->entries->numDocs > NR_MAXRANGE_SIZE && card > 1))
```
Also related to a doc and a record being equivalent.

Reply:
Fixed by commit [split NumericRange by InvertedIndex numEntries instead of numDocs](https://github.com/RediSearch/RediSearch/commit/a982d18c0f9a3a2dd23bc242882cd074c9ec5786)